mathesar/api/rpc/users/base.py

```diff
@@ -0,0 +1,43 @@
+from typing import Optional
+
+from django.contrib.auth import get_user_model
+from django.core.exceptions import ObjectDoesNotExist
+
+from mathesar.api.exceptions import APIError
```

Comment:
The `APIError` class is only for use in the (now deprecated) REST API. It's not a good idea here. You should raise a custom error class based on a standard Python `Exception` instead.